src/shared/bus-unit-util.c

```diff
@@ -865,6 +865,9 @@ int bus_deserialize_and_dump_unit_file_changes(sd_bus_message *m, bool quiet, Un
         const char *type, *path, *source;
         int r;
 
+        assert(changes);
+        assert(n_changes);
```

Comment:
`unit_file_dump_changes()` and `unit_file_changes_add()` are fine, but later we call
```
unit_file_dump_changes(0, NULL, *changes, *n_changes, false);
```
which dereferences `changes` with `*changes`. So yes, this is necessary.